server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -662,14 +656,20 @@ public void close() throws IOException {
     }
 
     /**
-     * @return Whether a field is a metadata field.
+     * @return Whether a field is a metadata field for older indexes
+     * This is an outdated method and should not be used.
+     * TODO: remove in v 9.0, as all DocumentField objects will contain information whether they are metaField or not
      */
-    public static boolean isMetadataField(String fieldName) {
-        return META_FIELDS.contains(fieldName);
+    public static boolean isMetadataFieldSt(String fieldName) {
```

Comment:
Does "St" stand for "static"? Interesting naming, no need to change, since the method shouldn't me used. Maybe add  @deprecated annotation to drive home the message?